R/dbi-methods.R

```diff
@@ -167,8 +170,13 @@ setMethod("dbSendQuery", c("SQLServerConnection", "character"),
     if (is_parameterised(ps) && is.null(params)) {
       return(pre_res)
     } else {
-      if (is_parameterised(ps) && !is.null(params)) {
-        dbBind(pre_res, params)
+      qry_nparams <- num_parameters(ps)
+      nparams <- length(params)
+      if (qry_nparams > 0 && nparams > 0L) {
+        if (nparams > qry_nparams) {
+          params <- params[1:qry_nparams]
+        }
+        dbBind(pre_res, params, batch = batch)
```

Comment:
Yes probably better to move to `dbBInd()`. Though I'm wondering whether we should be letting this fail in case when too many / too few are supplied.